apt/minimization/minimizer.py

```diff
@@ -121,7 +133,7 @@ def set_params(self, **params):
     def generalizations(self):
         return self.generalizations_
 
-    def fit_transform(self, X=None, y=None):
+    def fit_transform(self, X: Union[np.ndarray, pd.DataFrame] = None, y: Union[np.ndarray, pd.DataFrame] = None):
```

Comment:
Please update the description of the return value to indicate that it is a transformed numpy or pandas array (same as the transform method), and not self.